### PR TITLE
Compile scaled_dot_product_attention

### PR DESCRIPTION
**Pull request by cbalioglu**
With this PR, we now by default `torch.compile()` `torch.nn.functional.scaled_dot_product_attention` in `TorchSDPA`. This brings significant speed up for batches with padded inputs, which is typically the case for seq2seq models. I will run further benchmarks for non-padded inputs and see how well flash and memory_efficient attentions perform under `torch.compile()`.

### CODE REVIEW

**Comment by shagunsodhani** [comment]
Would love to know how these numbers look!

**Comment by cbalioglu** [comment]
Hey @shagunsodhani, I felt uneasy with this PR and why `torch.compile()` was able to speed up SDPA so much even though it was a highly optimized/fused implementation. Spent quite a bit of time last night and it turned out to be that providing a boolean mask instead of a float mask to `scaled_dot_product_attention` was significantly degrading its runtime performance. I have the "real fix" in #439 and removed this `torch.compile` statement in that PR.

**Comment by shagunsodhani** [comment]
Awesome! thank you @cbalioglu !

**Comment by cbalioglu** [comment]
You are welcome!
